sda-download/internal/config/config.go

```diff
@@ -479,24 +482,36 @@ func constructWhitelist(obj []TrustedISS) *jwk.MapWhitelist {
 }
 
 // GetC4GHKey reads and decrypts and returns the c4gh key
-func GetC4GHKey() (*[32]byte, error) {
-	log.Info("reading crypt4gh private key")
-	keyPath := viper.GetString("c4gh.filepath")
-	passphrase := viper.GetString("c4gh.passphrase")
+func GetC4GHKey() ([32]byte, string, error) {
+	log.Info("creating temporary crypt4gh key")
+
+	public, private, err := keys.GenerateKeyPair()
 
-	// Make sure the key path and passphrase is valid
-	keyFile, err := os.Open(keyPath)
 	if err != nil {
-		return nil, err
+		log.Errorf("Error when generating keys: %v", err)
+
+		return [32]byte{}, "", err
 	}
 
-	key, err := keys.ReadPrivateKey(keyFile, []byte(passphrase))
+	pem := bytes.Buffer{}
+	err = keys.WriteCrypt4GHX25519PublicKey(&pem, public)
+
 	if err != nil {
-		return nil, err
+		log.Errorf("Error when converting public key to PEM format: %v", err)
+
+		return [32]byte{}, "", err
 	}
 
-	keyFile.Close()
-	log.Info("crypt4gh private key loaded")
+	b64 := bytes.Buffer{}
+
+	encoder := base64.NewEncoder(base64.StdEncoding, &b64)
+	_, err = encoder.Write(pem.Bytes())
+
+	if err != nil {
+		log.Errorf("Error when converting public key to PEM format: %v", err)
+
+		return [32]byte{}, "", err
+	}
```

Comment:
for me `base64.StdEncoding.EncodeToString(publicKey[:])` worked as well here https://github.com/neicnordic/sda-download/pull/330/files#diff-bd6683a3964980b843bd0f1e474cb2ae7e33a91430a74f5b5770daad2983f0e7R306

Reply:
Name changed to `GenerateC4GHKey` and easier base64-ification is there now.